src/compiler/file_uploader/expand_paths.ts

```diff
@@ -0,0 +1,50 @@
+import { existsSync } from 'fs';
+import { readdir, stat } from 'fs/promises';
+import { join } from 'path';
+import { Src, Dest } from '.';
+
+export async function expandPaths(paths: [Src, Dest][]) {
+    return paths.reduce(
+        async (accPromise: Promise<[Src, Dest][]>, [srcPath, destPath]) => {
+            const acc = await accPromise;
+            return [...acc, ...(await expandPath(srcPath, destPath))];
+        },
+        Promise.resolve([])
+    );
+}
+
+async function expandPath(
+    srcPath: Src,
+    destPath: Dest
+): Promise<[Src, Dest][]> {
+    if (!existsSync(srcPath)) {
```

Comment:
Using `fs.exists` is deprecated. Instead, the current recommendation is to
directly attempt the file operation and handle errors with a `try...catch`
block. This approach eliminates the possibility of the file's existence state
changing between checking and performing the operation. Since we're throwing an
error if the file doesn't exist, I'm going to get rid of the throw and simply
let `fs` handle it for us.